tensorboard/webapp/settings_link_converter/_redux/settings_link_converter_types.ts

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+export const SETTINGS_LINK_CONVERTER_FEATURE_KEY = 'settingsLinkConverter';
+
+// TODO(jieweiwu): Update to a subset of RunsDataState and Metrics(None)NamespacedState,
+export type ShareableSettingState = null;
+export type SettingLinkConverterState = {
+  settings: ShareableSettingState;
```

Comment:
I don't see any need for there to be ShareableSettingState actually stored in ngrx State. I examined your prototype but couldn't see how storing this in the state is useful.

On the other hand, you do use ShareableSettingState as a way of passing data around (by your data source and by your actions) but then that's not "state", that's just a type. It might be better to just call it ShareableSettings or SharedSettings.